src/main/java/com/lothrazar/cyclic/block/uncrafter/ContainerUncraft.java

```diff
@@ -25,7 +25,15 @@ public ContainerUncraft(int windowId, World world, BlockPos pos, PlayerInventory
     ItemStackHandler outputSlots = tile.outputSlots;
     this.endInv = inputSlots.getSlots() + outputSlots.getSlots();
     //the main slot
-    addSlot(new SlotItemHandler(inputSlots, 0, 39, 19));
+    addSlot(new SlotItemHandler(inputSlots, 0, 39, 19) {
+
+      @Override
+      public void onSlotChanged() {
```

Comment:
This took ages to figure out. When a recipe needs multiple of an item, and you right-click to add more, the inventory change methods aren't always called. This forces them to be.